### DISCUSSION TITLE
Multiple table in one page

### DISCUSSION THREAD

**Discussion by mstfkhazaal** [filamentphp/filament#9670] [Help] [answered]
How can add multiple table in one page

**Comment by sinnbeck** [comment]
Panel or regular blade/livewire page ?

**Reply by mstfkhazaal** [reply]
Panel `https://filamentphp.com/docs/3.x/panels/pages`

**Reply by mstfkhazaal** [reply]
i have page with wizard form 
in every step i want to add table
i create custom Field with table 

**components.table.table_1.blade**
```
<div>
    {{ $this->table}}
</div>
```
**components.table.table_2.blade**
```
<div>
    {{ $this->table2 }}
</div>
```

**Page**
```
Form:

       Wizard::make([
               Wizard\Step::make('Step 1')
                        ->schema([
                            Components\ViewField::make('table_1')           
                                ->view('components.table.table_1'),
                        ]),
             Wizard\Step::make('Step 2')
                        ->schema([
                            Components\ViewField::make('table_2')           
                                ->view('components.table.table_2'),
                        ]),
                        ..
   ]),

    public function table(Table $table): Table
    {
        return $table
            ->columns([
                TextColumn::make('title'),
            ])
            ->query(function (){
                return User::query();
            });
    }
  public function table2(Table $table): Table
    {
        return $table
            ->columns([
                TextColumn::make('title'),
            ])
            ->query(function (){
                return User::query();
            });
    }
```
i get this error
App\Filament\Organization\Pages\Admin\VerifyAssessment::table2: Argument #1 ($table) must be of type Filament\Tables\Table, Filament\Infolists\Infolist given, called in /var/www/html/vendor/filament/infolists/src/Concerns/InteractsWithInfolists.php on line 55

**Reply by leandrocfe** [reply]
[Create two components and add the table](https://filamentphp.com/docs/3.x/tables/adding-a-table-to-a-livewire-component/#setting-up-the-livewire-component)

Render in each view

@livewire('table1')

@livewire('table2')

**Reply by mstfkhazaal** [reply]
is work but i need tables inside wizard component how?

**Reply by leandrocfe** [reply]
Do you have this view

`Components\ViewField::make('table_1')                           ->view('components.table.table_1')`

Did you try to render @livewire('table1') inside this view?

**Reply by mstfkhazaal** [reply]
if render in the view is worked successfully

but i need to add inside the form (Wizard)
i tried
Components\ViewField::make('table_1')->view('components.table.table_1')
i get this error 
Property [$table] not found on component:

**Reply by mstfkhazaal** [reply]
and i tried use custom field i get same error

**Reply by leandrocfe** [reply]
![image](https://github.com/filamentphp/filament/assets/3833889/d293e648-3498-4434-81f8-38c9b9184adc)

Did you follow this setup?
https://filamentphp.com/docs/3.x/tables/adding-a-table-to-a-livewire-component#adding-the-table

You should do 

```php
<div>
    {{ $this->table }}
</div>
```
in the component view

**Reply by mstfkhazaal** [reply]
this work but how can add another table in second step ?

**Reply by leandrocfe** [reply]
The same process 🤷‍♂️
You need to create two livewire compone ts. Table1 and table2

**Reply by mstfkhazaal** [reply]
i get this error
Property [$table] not found on component:

**Reply by mstfkhazaal** [reply]
i added Table inside Custom Field
but i get this error?
`Property [$table] not found on component`
https://github.com/filamentphp/filament/discussions/4607

**Reply by leandrocfe** [reply]
Something is wrong because it works as in my screenshot. Please share your code on GitHub, and I can take a look for you.

**Reply by mstfkhazaal** [reply]
https://github.com/mstfkhazaal/test-table-field

this repo 

Issue: https://github.com/filamentphp/filament/issues/9695

**Reply by leandrocfe** [reply, marked as answer]
What is the problem? You didn't follow [these steps](https://filamentphp.com/docs/3.x/tables/adding-a-table-to-a-livewire-component#adding-the-table)
First of all, create two livewire components

`php artisan make:livewire Table1`
`php artisan make:livewire Table2`

2) [add the table](https://filamentphp.com/docs/3.x/tables/adding-a-table-to-a-livewire-component#adding-the-table) in `app/Livewire/Table1.php` and `app/Livewire/Table2.php`

```html
<!-- resources/views/livewire/table1.blade.php -->
<div>
    {{ $this->table }}
</div>
```

```html
<!-- resources/views/livewire/table2.blade.php -->
<div>
    {{ $this->table }}
</div>
```

3) render the table in the view field

```html
<!-- resources/views/forms/components/table1.blade.php-->
<div>
@livewire('table1')
</div>
```

```html
<!-- resources/views/forms/components/table2.blade.php-->
<div>
@livewire('table2')
</div>
```

**Reply by mstfkhazaal** [reply]
Thanks
The reason is on me, I did not understand the document well
